app/src/main/java/no/usn/mob3000/domain/viewmodel/AuthenticationViewModel.kt

```diff
@@ -90,8 +92,39 @@ class LoginViewModel(
             logoutUseCase()
         }
     }
+
+    /**
+     * Initiates the registration process with the provided credentials.
+     *
+     * @param email The user's e-mail address.
+     * @param password The user's password.
+     * @author Anarox
+     * @created 2024-10-30
+     */
+
+    fun register(email: String, password: String) {
+        viewModelScope.launch {
+            _authenticatedUser.value = null
+            _loginState.value = LoginState.Loading
+            registerUseCase(email, password).fold(
+                onSuccess = { user ->
+                    _authenticatedUser.value = user
+                    _loginState.value = LoginState.Success(user)
+                },
+                onFailure = { error ->
+                    val authError = when (error) {
+                        is AuthRestException -> AuthError.fromException(error)
+                        else -> AuthError.Unknown(error.message ?: "Unknown error occurred")
+                    }
```

Comment:
`AuthError` is used and designed for login's use-case. You are unlikely to get a usable error message, if any, when trying to register an account. If there are multiple different registration event responses like there are for authentication, then registration should have its own.

Reply:
Will look into that.